File: C++/83_Remove_Duplicates_from_Sorted_List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(head == NULL) return NULL;
        auto t = head;
        auto l = head->next;
        while(t != NULL){
            if(l == NULL){
                t->next = NULL;
                break;
            }
            if(t->val == l->val){
                l = l->next;
            } else {
                t->next = l;
                t = t->next;
            }
        }
        return head;
    }
};

File: Python/83_Remove_Duplicates_from_Sorted_List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        if head == None:
            return None
        t = head
        l = head.next
        while t != None:
            if l == None:
                t.next = None
                break
            if t.val == l.val:
                l = l.next
            else:
                t.next = l
                t = t.next
        return head